### PR TITLE
adds lti launch routes in dev environment

### PR DESCRIPTION
**Pull request by iturgeon**
adds the following routes when webpack-dev-server is active:

* /dev (links to all the things below)
* /routes
* /lti/dev/launch/course_navigation
* /lti/dev/launch/view
* /lti/dev/launch/resource_selection
* /lti/dev/return/resource_selection

### CODE REVIEW

File: packages/app/obojobo-express/obo_express_dev.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+			.sort((a,b) => a.path.localeCompare(b.path))
+
+		res.json(simplifiedEndpoints)
+	})
+
+	// builds a valid course navigation lti launch and submits it
+	app.get('/lti/dev/launch/course_navigation', (req, res) => {
+		const method = 'POST'
+		const endpoint = `${baseUrl(req)}/lti/canvas/course_navigation`
+		const params = {
```
Comment (gitnix) [inline packages/app/obojobo-express/obo_express_dev.js]: would suggest making a common params object that contains user info and spreading over that inside this and subsequent route params

File: packages/app/obojobo-express/webpack.config.js [hunk 2]
```diff
@@ -44,6 +45,7 @@ module.exports =
 				watchOptions: {
 					ignored: '/node_modules/'
 				},
+				setup: ObojoboExpressDev,
```
Comment (gitnix) [inline packages/app/obojobo-express/webpack.config.js]: The setup option is deprecated. Suggest moving into `before` function.

Comment (zachberry) [reply packages/app/obojobo-express/webpack.config.js]: #busted

File: packages/app/obojobo-express/obo_express_dev.js (hunk 1 above)
Comment (zachberry) [reply packages/app/obojobo-express/obo_express_dev.js]: Agreed

**Comment by iturgeon** [comment]
I'd argue for that if their values would ever be used anywhere else.  IMO, they have no use anywhere but in this page - and they are unlikely to be something you want to change given the use case for these pages.

**Pull request merged**